streams/src/test/java/org/apache/kafka/streams/kstream/JoinWindowsTest.java

```diff
@@ -117,6 +118,13 @@ public void gracePeriodShouldEnforceBoundaries() {
         }
     }
 
+    @Test
+    public void oldAPIShouldSetDefaultGracePeriod() {
```

Comment:
I think you also need the following test (here and in the other `*Windows`):
```
assertThat(DEPRECATED_OLD_24_HR_GRACE_PERIOD, is(Duration.ofDays(1).toMillis()));
```
Otherwise we will not notice whether `DEPRECATED_OLD_24_HR_GRACE_PERIOD` is changed by mistake.

Reply:
Alternatively, you could use `24 * 60 * 60 * 1000L` instead of `DEPRECATED_OLD_24_HR_GRACE_PERIOD` in the tests.